keps/sig-apps/20190318-Graduate-CronJob-to-Stable.md

```diff
@@ -0,0 +1,579 @@
+---
+title: graduate-cronjob-to-stable  
+authors:  
+  - "@barney-s"  
+
+owning-sig: sig-apps  
+participating-sigs:
+  - sig-scalability
+
+reviewers:
+  - "@liggitt"
+  - "@kow3ns"
+  - "@janetkuo"
+  - “@mortent”
+  - "@wojtek-t"
+
+approvers:
+  - "@kow3ns"
+  - "@liggitt"
+  - "@soltysh"
+  - "@wojtek-t"
+
+editor: TBD  
+creation-date: 2019-04-18  
+last-updated: 2019-04-18  
+status: provisional  
+see-also:  
+replaces:  
+superseded-by:  
+
+---
+
+# Graduate CronJob to stable
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Goals](#goals)
+- [Proposal](#proposal)
+  - [Develop new CronJob controller](#develop-new-cronjob-controller)
+    - [Existing controller](#existing-controller)
+    - [Proposed rearchitect](#proposed-rearchitect)
+      - [Informers and Caches](#informers-and-caches)
+      - [Multiple workers](#multiple-workers)
+      - [Handling Cron aspect](#handling-cron-aspect)
+      - [Metrics](#metrics)
+  - [Add .status.lastSuccessfulTime](#add-statuslastsuccessfultime)
+  - [Add .status.nextScheduleTime](#add-statusnextscheduletime)
+  - [Add Counters](#add-counters)
+  - [Add .status.conditions](#add-statusconditions)
+  - [Support Jitter for cronjobs](#support-jitter-for-cronjobs)
+  - [Fix applicable open issues](#fix-applicable-open-issues)
+  - [Scale Targets for GA](#scale-targets-for-ga)
+    - [CronJob Limits](#cronjob-limits)
+    - [Frequency of launched jobs](#frequency-of-launched-jobs)
+- [V1 API](#v1-api)
+- [v1beta1 changes](#v1beta1-changes)
+- [Validations](#validations)
+- [Plan for promoting CronJob to GA](#plan-for-promoting-cronjob-to-ga)
+- [Tests](#tests)
+  - [E2E test](#e2e-test)
+    - [Existing test cases](#existing-test-cases)
+    - [New test cases](#new-test-cases)
+  - [Conformance Tests](#conformance-tests)
+  - [Unit Tests](#unit-tests)
+- [Implementation plan](#implementation-plan)
+    - [Release X](#release-x)
+    - [Release X+1](#release-x1)
+    - [Release X+2](#release-x2)
+- [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Summary
+
+[CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) is a Kubernetes API that creates Job object on a schedule specified by a cron spec. It is in beta status since v1.8. This document lays out the plan to promote it to stable.
+
+## Motivation
+
+CronJob definition has been stable for the last few releases and is useful to run periodic tasks in kubernetes cluster. This API adds the ability to add cron facility to a cluster. We feel the API is ready to be promoted to Stable and be supported longterm by the community.
```

Comment:
Re API: it seems we want to extend it, so "it's ready to be promoted to stable" is misleading. You should make that explicit that we believe we know how it should look like but it's not exactly the current API.

Reply:
The changes in API we are proposing is only in `.status` . We are not changing `.spec`

Reply:
reworded to show intent to change `.status`